Mathlib/AlgebraicTopology/SimplicialSet/Coskeletal.lean

```diff
@@ -0,0 +1,270 @@
+    (s : Cone (proj (op [n]) (Truncated.inclusion 2).op ⋙ (Truncated.inclusion 2).op ⋙ X))
+    (x : s.pt)
+    (φ : [1] ⟶ [n]) :
+    X.map φ.op (lift s x) = s.π.app (strArrowMk₂ φ (by omega)) x := by
+  obtain ⟨i, j, hij, rfl⟩ : ∃ i j hij, φ = mkOfLe i j hij :=
+    ⟨φ.toOrderHom 0, φ.toOrderHom 1, φ.toOrderHom.monotone (by simp),
+      Hom.ext_one_left _ _ rfl rfl⟩
+  exact fac_aux₂ _ _ _ _ _ _ (by omega)
+
+end isPointwiseRightKanExtensionAt
+
+open Truncated
+
+open isPointwiseRightKanExtensionAt in
+/-- A strict Segal simplicial set is 2-coskeletal. -/
+noncomputable def isPointwiseRightKanExtensionAt (n : ℕ) :
+    (rightExtensionInclusion X 2).IsPointwiseRightKanExtensionAt ⟨[n]⟩ where
+  lift s x := lift (X := X) s x
+  fac s j := by
+    ext x
+    obtain ⟨⟨i, hi⟩, ⟨f⟩, rfl⟩ := j.mk_surjective
+    obtain ⟨i, rfl⟩ : ∃ j, SimplexCategory.mk j = i := ⟨_, i.mk_len⟩
+    change [i] ⟶ [n] at f
+    dsimp at hi ⊢
+    apply StrictSegal.spineInjective
+    dsimp
+    ext k
+    · dsimp only [spineEquiv, Equiv.coe_fn_mk]
+      erw [spine_map_vertex]
+      rw [spine_spineToSimplex, spine_vertex]
+      let α : strArrowMk₂ f hi ⟶ strArrowMk₂ ([0].const [n] (f.toOrderHom k)) (by omega) :=
+        StructuredArrow.homMk (([0].const _ (by exact k)).op) (by simp; rfl)
+      exact congr_fun (s.w α).symm x
+    · dsimp only [spineEquiv, Equiv.coe_fn_mk]
+      dsimp
+      rw [← FunctorToTypes.map_comp_apply]
+      let α : strArrowMk₂ f hi ⟶ strArrowMk₂ (mkOfSucc k ≫ f) (by omega) :=
+        StructuredArrow.homMk (mkOfSucc k).op (by simp; rfl)
+      exact (isPointwiseRightKanExtensionAt.fac_aux₃ _ _ _ _).trans (congr_fun (s.w α).symm x)
+  uniq s m hm := by
+    ext x
+    apply StrictSegal.spineInjective (X := X)
+    dsimp [spineEquiv]
+    rw [StrictSegal.spine_spineToSimplex]
+    ext i
+    · exact congr_fun (hm (StructuredArrow.mk (Y := op [0]₂) ([0].const [n] i).op)) x
+    · exact congr_fun (hm (.mk (Y := op [1]₂) (.op (mkOfLe _ _ (Fin.castSucc_le_succ i))))) x
+
+/-- Since `rightExtensionInclusion₂IsPointwiseRightKanExtensionAt X n` proves that the appropriate
+cones are limit cones, `rightExtensionInclusion X 2` is a pointwise right Kan extension.-/
+noncomputable def isPointwiseRightKanExtension :
+    RightExtension.IsPointwiseRightKanExtension (rightExtensionInclusion X 2) :=
+  fun Δ => isPointwiseRightKanExtensionAt X Δ.unop.len
+
+theorem isRightKanExtension :
+    (rightExtensionInclusion X 2).left.IsRightKanExtension (rightExtensionInclusion X 2).hom :=
+  RightExtension.IsPointwiseRightKanExtension.isRightKanExtension
+    (isPointwiseRightKanExtension X)
+
+/-- When `X` is `StrictSegal`, `X` is 2-coskeletal. -/
+instance isCoskeletal : SimplicialObject.IsCoskeletal X 2 where
+  isRightKanExtension := isRightKanExtension X
+
+/-- The canonical isomorphism `X ≅ (cosk 2).obj X` defined when `X` is `StrictSegal`. -/
+noncomputable def isoCosk2 : X ≅ (cosk 2).obj X := SimplicialObject.isoCoskOfIsCoskeletal X 2
+
```

Comment:
This can be removed: because of the typeclass system, when `X` is strict Segal, it is 2-coskeletal due to the instance above, and then the isomorphism is just `X.isoCoskOfIsCoskeletal 2`.
```suggestion
```